### PR TITLE
feat(`datree`): add completion

### LINKED ISSUE #1302: [datree]
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [X] I would like to work on this.

### What CLI tool does this relate to?

datree

### Which statement makes the most sense?

This completion spec does not exist

### Issue Details

- Add `datree` CLI

## Datree
Github: [datreeio/datree](https://github.com/datreeio/datree)
Web: [datree.io](https://datree.io)

### PR DESCRIPTION
**Pull request by l2D**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
- feature:
    - Add `datree`

**What is the current behavior? (You can also link to an open issue here)**
- Closes #1302 

**What is the new behavior (if this is a feature change)?**
- `datree` CLI completion

**Additional info:**

### CODE REVIEW

File: src/datree.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
+const testOptions: Fig.Option[] = [
+  {
+    name: ["--help", "-h"],
+    description: "Helo for 'test'",
```
Comment (fedeci) [inline src/datree.ts:4]:
```suggestion
    description: "Help for 'test'",
```

File: src/datree.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      description: "Generate completion script for bash,zsh,fish,powershell",
+    },
+    {
+      name: "config",
+      description: "Internal configuration management for datree config file",
+      subcommands: [
+        {
+          name: "get",
+          description: "Get configuration value",
+          additionalSuggestions: ["--help"],
```
Comment (fedeci) [inline src/datree.ts:100]:
```suggestion
```
This is not required

File: src/datree.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      options: [
+        {
+          name: ["-h", "--help"],
+          description: "Help for config",
+        },
+      ],
+    },
+    {
+      name: "help",
+      description: "Help about any command",
```
Comment (fedeci) [inline src/datree.ts:131]:
```suggestion
      description: "Help about any command",
      args: {
        template: 'help',
      },
```

**Review by fedeci — APPROVED** [review]
Thanks, there are just a couple nits to fix before merging it!

File: src/datree.ts (hunk 1 above)
Comment (l2D) [reply src/datree.ts:4]: Fixed.

File: src/datree.ts (hunk 2 above)
Comment (l2D) [reply src/datree.ts:100]: Fixed

File: src/datree.ts (hunk 3 above)
Comment (l2D) [reply src/datree.ts:130]: - Fix `'help'` to `"help"` on [64d8df6](https://github.com/withfig/autocomplete/pull/1303/commits/64d8df6d84017ef6b1320bea7b56192f20c0cf06)

File: src/datree.ts (hunk 2 above)
Comment (fedeci) [reply src/datree.ts]: There are other occurrences of this "issue" in the file, can you fix those also?

Comment (l2D) [reply src/datree.ts]: Sure, thanks. 👍

Comment (l2D) [reply src/datree.ts]: - Update on [66eee14](https://github.com/withfig/autocomplete/pull/1303/commits/66eee146ee6bbf1caa7c35330ce6903d6825f7a9)

**Comment by fedeci** [comment]
Published

**Pull request merged**